File: android_app/src/main/java/me/timeto/app/ui/ActivityPickerSheet.kt
package me.timeto.app.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import me.timeto.app.c
import me.timeto.app.rememberVm
import me.timeto.shared.db.ActivityDb
import me.timeto.shared.vm.ActivityPickerSheetVm

@Composable
fun ActivityPickerSheet(
    layer: WrapperView.Layer,
    onPick: (ActivityDb) -> Unit,
) {

    val (_, state) = rememberVm { ActivityPickerSheetVm() }

    Column(
        modifier = Modifier
            .fillMaxHeight()
            .background(c.sheetBg)
    ) {

        val scrollState = rememberScrollState()

        Sheet.HeaderViewOld(
            onCancel = { layer.close() },
            title = state.headerTitle,
            doneText = null,
            isDoneEnabled = false,
            scrollState = scrollState,
            cancelText = "Back",
        ) {}

        Column(
            modifier = Modifier
                .verticalScroll(state = scrollState)
                .padding(bottom = 20.dp)
                .navigationBarsPadding()
                .imePadding()
        ) {

            Row(Modifier.height(20.dp)) { }

            val activitiesUI = state.activitiesUI
            activitiesUI.forEach { activityUI ->
                val isFirst = state.activitiesUI.first() == activityUI
                MyListView__ItemView(
                    isFirst = isFirst,
                    isLast = state.activitiesUI.last() == activityUI,
                    withTopDivider = !isFirst,
                ) {
                    MyListView__ItemView__ButtonView(
                        text = activityUI.text,
                    ) {
                        onPick(activityUI.activity)
                        layer.close()
                    }
                }
            }
        }
    }
}

File: ios_app/iphone/ActivityPickerSheet.swift
import SwiftUI
import shared

struct ActivityPickerSheet: View {

    @Binding var isPresented: Bool
    let onPick: (ActivityDb) -> Void

    @State private var vm = ActivityPickerSheetVm()
    @State private var sheetHeaderScroll = 0

    var body: some View {

        VMView(vm: vm, stack: .VStack(spacing: 0)) { state in

            SheetHeaderView(
                    onCancel: { isPresented.toggle() },
                    title: state.headerTitle,
                    doneText: nil,
                    isDoneEnabled: false,
                    scrollToHeader: sheetHeaderScroll,
                    cancelText: "Back"
            ) {}

            ScrollViewWithVListener(showsIndicators: false, vScroll: $sheetHeaderScroll) {

                VStack(spacing: 0) {

                    let activitiesUI = state.activitiesUI
                    ForEach(activitiesUI, id: \.activity.id) { activityUI in

                        let isFirst = activitiesUI.first == activityUI

                        MyListView__ItemView(
                                isFirst: isFirst,
                                isLast: activitiesUI.last == activityUI,
                                withTopDivider: !isFirst
                        ) {

                            MyListView__ItemView__ButtonView(
                                    text: activityUI.text
                            ) {
                                onPick(activityUI.activity)
                                isPresented = false
                            }
                        }
                    }
                }
                        .padding(.top, 20)
            }
        }
                .background(c.sheetBg)
    }
}
